src/Microsoft.VisualStudio.Composition/Configuration/Formatters/MessagePackSerializerContext.cs

```diff
@@ -0,0 +1,161 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.VisualStudio.Composition
+{
+    using System.Collections.Concurrent;
+    using System.Diagnostics;
+    using System.Diagnostics.CodeAnalysis;
+    using System.Reflection;
+    using System.Runtime.Serialization;
+    using MessagePack;
+    using MessagePack.Formatters;
+    using MessagePack.Resolvers;
+    using Microsoft.VisualStudio.Composition.Formatter;
+
+    internal class MessagePackSerializerContext : MessagePackSerializerOptions, IDisposable
+    {
+        public MessagePackSerializerContext(IFormatterResolver resolver, Resolver compositionResolver)
+            : base(GetIFormatterResolver(resolver))
+        {
+            this.CompositionResolver = compositionResolver;
+        }
+
+        public Resolver CompositionResolver { get; }
+
+        public void Dispose()
+        {
+            GC.SuppressFinalize(this);
+        }
+
```

Comment:
This doesn't seem like a reasonable cause for being disposable in the first place.